controller/main_controller.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ *
+ * Collapsible Categories extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) 2015 phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\collapsiblecategories\controller;
+
+class controller implements main_interface
+{
+	/** @var \phpbb\config\config */
+	protected $config;
+
+	/** @var \phpbb\controller\helper */
+	protected $helper;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var string phpBB root path */
+	protected $root_path;
+
+	/** @var string phpEx */
+	protected $php_ext;
+
+	/**
+	* Constructor
+	*
+	* @param \phpbb\config\config                $config             Config object
+	* @param \phpbb\controller\helper            $helper             Controller helper object
+	* @param \phpbb\boardrules\operators\rule    $rule_operator      Rule operator object
+	* @param \phpbb\template\template            $template           Template object
+	* @param \phpbb\user                         $user               User object
+	* @param string                              $root_path          phpBB root path
+	* @param string                              $php_ext            phpEx
+	* @access public
+	*/
+	public function __construct(\phpbb\config\config $config, \phpbb\controller\helper $helper, \phpbb\template\template $template, \phpbb\user $user, $root_path, $php_ext)
+	{
+		$this->config = $config;
+		$this->helper = $helper;
+		$this->template = $template;
+		$this->user = $user;
+		$this->root_path = $root_path;
+		$this->php_ext = $php_ext;
```

Comment:
none of these are needed except for request, and our operator class
`\phpbb\collapsiblecategories\operator\operator`